framework/src/st_things/things_stack/things_stack.c

```diff
@@ -450,7 +427,7 @@ int things_reset(void *remote_owner, things_es_enrollee_reset_e resetType)
 			res = -1;
 			goto GOTO_OUT;
 		}
-		pthread_detach(h_thread_things_reset);
```

Comment:
Reason: Avoid detaching the reset thread to handle it during st_things_deinitialize().
If we detach it here, then there would be thread sync problem if app calls st_things_deinitialize().